### PR TITLE
extract EditorService#editors into subscribable, next, and value for easier future refactoring

### PR DESCRIPTION
**Pull request by sqs**
Refactor. This will let us make the subscribable merge in data from other sources (while still keeping the next/value APIs only dealing with normalized data).

This is the 1st PR in the sequence #3320 #3322 #3323  (review in order).

### CODE REVIEW

File: shared/src/api/client/services/editorService.ts [hunk 1]
```diff
@@ -16,27 +16,33 @@ export interface CodeEditorData<D extends Pick<TextDocument, 'uri'> = TextDocume
...
-export interface EditorService extends ReadonlyEditorService {
+export interface EditorService {
     /** All code editors. */
-    readonly editors: Subscribable<readonly CodeEditorData[]> & { value: readonly CodeEditorData[] } & NextObserver<
-            readonly CodeEditorData[]
-        >
+    readonly editors: Subscribable<readonly CodeEditorData[]>
+
+    /** Transitional API for synchronously getting the list of code editors. */
+    readonly editorsValue: readonly CodeEditorData[]
```
Comment (felixfbecker) [inline shared/src/api/client/services/editorService.ts]: In the API, we have these swapped (`editorChanges` and `editors`)

Comment (sqs) [reply shared/src/api/client/services/editorService.ts]: This is just an intermediate commit. This transitional API is removed immediately in the next 2 stacked PRs. So if you are OK with keeping it like this in this commit, that would save some work.

Comment (felixfbecker) [reply shared/src/api/client/services/editorService.ts]: Then maybe one PR would be better? It seems like I can't review a PR without the context of the following PRs

**Comment by sqs** [comment]
Abandoning in favor of https://github.com/sourcegraph/sourcegraph/pull/3322, which has this commit and one other stacked one that adds useful context to this PR's changes.

File: shared/src/api/client/services/editorService.ts (hunk 1 above)
Comment (sqs) [reply shared/src/api/client/services/editorService.ts]: Cool, see https://github.com/sourcegraph/sourcegraph/pull/3322

**Pull request closed without merging**